src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -812,7 +812,7 @@ public Descriptor getDescriptor()
     @Override
     public Path getFile()
     {
-        return descriptor.getDirectory().resolve(getFilename());
+        return descriptor.pathFor(Component.DATA);
```

Comment:
I'm not sure how we survived without the ability to resolve file via `getFile` for remote file system.

Reply:
The main issue is that: during flush, it won't preload flushed sstable because it `doesn't exist locally` and won't delete it locally because of wrong path.